core/src/size.rs

```diff
@@ -34,6 +34,22 @@ impl Size {
             height: self.height + padding.vertical() as f32,
         }
     }
+
+    /// Returns the minimum of each component of this size and another
+    pub fn min(self, other: Self) -> Self {
```

Comment:
It could be worth to use [euclid](https://lib.rs/crates/euclid) for geometric primitives. It also allow to create different types for different coordinate systems in order to not mix them.